src/azul/__init__.py

```diff
@@ -48,8 +48,9 @@ def num_dss_workers(self) -> int:
         return int(os.environ['AZUL_DSS_WORKERS'])
 
     def qualified_resource_name(self, resource_name):
-        prefix = os.environ['AZUL_RESOURCE_PREFIX']
-        return f"{prefix}{resource_name}-{self.deployment_stage}"
+        prefix = os.environ['AZUL_PREFIX']
+        self.validate_terms((prefix, self.deployment_stage), "-")
```

Comment:
Let's validate the deployment stage name in `.deployment_stage()`. That would simplify the signature of `validate_terms`. The error message should include the name of the offending environment variable. And let's be very strict: the term should match this regex:

`[a-z][a-z0-9]{2,14}`